sssd_test_framework/utils/virtual_smartcard.py

```diff
@@ -0,0 +1,205 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from pytest_mh import MultihostHost, MultihostUtility
+
+if TYPE_CHECKING:
+    from ..roles.client import Client
+
+__all__ = [
+    "SmartCardUtils",
+]
+
+
+class SmartCardUtils(MultihostUtility[MultihostHost]):
+    """
+    Utility class for managing smart card operations using SoftHSM and PKCS#11.
+    """
+
+    SOFTHSM2_CONF_PATH = "/opt/test_ca/softhsm2.conf"
+    TOKEN_STORAGE_PATH = "/opt/test_ca/tokens"
+    OPENSC_CACHE_PATHS = [
+        "$HOME/.cache/opensc/",
+        "/run/sssd/.cache/opensc/",
+    ]
+
+    def __init__(self, host: MultihostHost) -> None:
+        super().__init__(host)
+
+    def init(self, label: str = "sc_test", so_pin: str = "12345678", user_pin: str = "123456") -> None:
+        """
+        Initializes a SoftHSM token with the given label and PINs.
+
+        Cleans cache directories and prepares the token directory.
+
+        :param label: Token label, defaults to "sc_test"
+        :type label: str, optional
+        :param so_pin: Security Officer PIN, defaults to "12345678"
+        :type so_pin: str, optional
+        :param user_pin: User PIN, defaults to "123456"
+        :type user_pin: str, optional
+        """
+        for path in self.OPENSC_CACHE_PATHS:
```

Comment:
We have methods in pytest-mh to manipulate files and directories. 

https://pytest-mh.readthedocs.io/en/latest/articles/bundled-utilities/fs.html